src/main/java/org/opensearch/ad/ml/EntityColdStarter.java

```diff
@@ -313,7 +261,88 @@ private void coldStart(
             return;
         }
 
-        nodeStateManager.getAnomalyDetector(detectorId, onGetDetector(modelId, entity, detectorId, modelState, listener));
+        boolean earlyExit = true;
```

Comment:
This block seems to be just a shift of the existing block. Is that correct?

Reply:
yeah, I moved the enclosing logic of getting a detector to the caller, thus the callee does not need to do that (and you saw a shift of existing block).